estate_account/models/estate_property.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from odoo import models, Command
+
+
+class EstateAccountPropertyModel(models.Model):
+    _inherit = "estate.property"
+
+    def mark_as_sold(self):
+        self.ensure_one()
+        self.env["account.move"].create({
+            "name": self._generate_invoice_name(),
+            "partner_id": self.salesperson.id,
+            "move_type": "out_invoice",
+            "line_ids": [
+                Command.create({
+                    "name": self.name,
+                    "quantity": 1,
+                    "price_unit": self.selling_price
+                }),
+                Command.create({
+                    "name": "VAT",
+                    "quantity": 1,
+                    "price_unit": self.selling_price * .06
+                }),
+                Command.create({
+                    "name": "Administrative Fees",
+                    "quantity": 1,
+                    "price_unit": 100.00
+                })
+            ]
+        })
+        return super().mark_as_sold()
+
+    def _generate_invoice_name(self):
+        taken_ids = set()
+        for invoice in self.env["account.move"].search([]):
+            if (match := re.match(r"Invoice\s+(\d+)", invoice.name)):
+                taken_ids.add(int(match.group(1)))
+        i = 1
+        while i in taken_ids:
+            i += 1
+        return f"Invoice {i}"
```

Comment:
Do you want to select the first number missing in the sequence or the one above the maximum ?
If it is the last case, I would do something like that to avoid too many loops.
```suggestion
        invoice = self.env["account.move"].search([("name", "like", "Invoice %")], order='id desc', limit=1)
        if invoice and (match := re.match(r"Invoice (\d+)", invoice.name)):
            return f"Invoice {match.group(1)}"
        return "Invoice 1"
```